app/javascript/components/bookmarks/BookmarkManager.jsx

```diff
@@ -0,0 +1,286 @@
+import { Popover, OverlayTrigger } from 'react-bootstrap'
+import BookmarksList from '~/components/bookmarks/BookmarksList'
+import { fetchBookmarks, createBookmark, updateBookmark, deleteBookmark } from '~/lib/scp-api'
+import React, { useEffect, useState } from 'react'
+import _cloneDeep from 'lodash/clone'
+import { isUserLoggedIn } from '~/providers/UserProvider'
+import { useLocation } from '@reach/router'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faLink, faUndo } from '@fortawesome/free-solid-svg-icons'
+import useErrorMessage from '~/lib/error-message'
+
+/**
+ * Component to bookmark views in Explore tab, as well as existing link/reset buttons
+ *
+ * @param bookmarks {Array} existing user bookmark objects
+ * @param clearExploreParams
+ */
+export default function BookmarkManager({bookmarks, clearExploreParams}) {
+  const location = useLocation()
+  const [allBookmarks, setAllBookmarks] = useState(bookmarks)
+  const [saveText, setSaveText] = useState('Save')
+  const [saveDisabled, setSaveDisabled] = useState(false)
+  const [bookmarkSaved, setBookmarkSaved] = useState(null)
+  const [deleteDisabled, setDeleteDisabled] = useState(false)
+  const { ErrorComponent, setShowError, setError } = useErrorMessage()
+  const DEFAULT_BOOKMARK = {
+    name: '',
+    description: '',
+    path: getBookmarkPath()
+  }
+
+  /** copies the url to the clipboard */
+  function copyLink() {
+    navigator.clipboard.writeText(location.href)
+  }
+
+  /** Find a matching bookmark using the current URL params, or return default empty bookmark  */
+  function findExistingBookmark() {
+    const bookmark = allBookmarks.find(bookmark => { return bookmark.path === getBookmarkPath() })
```

Comment:
More conventional, simpler:

```suggestion
    const bookmark = allBookmarks.find(bookmark => bookmark.path === getBookmarkPath())
```

For reference:
```
['a', 'b', 'c'].find(letter => { return letter === 'b' })
'b'
['a', 'b', 'c'].find(letter => letter === 'b')
'b'
```

Reply:
Very nice!

Reply:
[309f02a](https://github.com/broadinstitute/single_cell_portal_core/pull/1979/commits/309f02ac5945957b77172d2b36feef535c4b063b)